### PR TITLE
Makes luxury shuttle more user friendly

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

People can't seem to work out how to use the shuttle and hate it for that reason. This makes it easier for them.

## Why It's Good For The Game

Hopefully more people will be able to board the first class area of the luxury shuttle and appreciate it.

## Changelog
:cl: cacogen
add: Luxury shuttle now tells you if you don't have arms to drag money into the ticket collector, which will hopefully teach players who don't know to do this
add: Luxury shuttle can now access accounts on dragged ID cards
tweak: Change doesn't disappear if the payee isn't human or doesn't have a bank account
tweak: Change is now placed in the payee's hand, and is otherwise placed on the ground with the payee dragging it if their hands are full or they don't have any
tweak: Luxury shuttle now tells you privately how much more money you need, for less spam. It will still loudly welcome you aboard in front of your co-workers.
/:cl:

**I've edited this since these drive-by thumbs so change is placed in-hand for humans instead.**

### CODE REVIEW

**Comment by FantasticFwoosh** [comment]
If there isn't a proletariat uprising against the captain as soon as the shuttle is bought and locked in, or people start taking additional work to pay off their fare, the shuttles's gimmick would be wasted with this PR.

Please remove the spare change feature for something else, like the heads altruistically paying off fares to assist crewmembers boarding out of their own pocket at shuttle console. :+1: for all the other changes.

**Comment by Dawson1917** [comment]
tweak: Luxury shuttle now drops change at the feet of people boarding for people without enough money to take

Everything is good besides this. This shuttle is, essentially, for griff and is funny for it. Locking people in the back while all the heads of staff and miners and assistants who stole from ID cards ride in the front compartment is the point. Heads paying out of their pocket for the sake of letting everyone else not ride in the mule den goes against the theme of the shuttle, in that all the heads are selfish and couldn't care less for the rest of the crew's conditions so long as they get their champagne and caviar.

**Comment by bgobandit** [comment]
space communism :+1:

**Comment by cacogen** [comment]
>If there isn't a proletariat uprising against the captain as soon as the shuttle is bought and locked in, or people start taking additional work to pay off their fare, the shuttles's gimmick would be wasted with this PR.

This doesn't happen

>Please remove the spare change feature for something else

I'll put it in their hands instead, if only for the people giving this downward facing thumbs

**Comment by Farquaar** [comment]
I'd echo the sentiments of others here. It's more fun when players actively have to give their fellow crewmembers money if they want to the poor working stiffs to ride in the luxury area. It facilitates more player interaction and roleplay that way.

File: code/modules/shuttle/special.dm [hunk 1]
```diff
@@ -262,40 +284,50 @@
 		payees[AM] += H.credits
 		counted_money += H
 
-	if(ishuman(AM))
-		var/mob/living/carbon/human/H = AM
-		if(H.get_bank_account())
-			var/datum/bank_account/account = H.get_bank_account()
+	if(payees[AM] < threshold)
+		var/armless
+		if(!ishuman(AM) && !istype(AM, /mob/living/simple_animal/slime))
```
Comment (actioninja) [inline code/modules/shuttle/special.dm]: Is this intended to be not slime?

Comment (cacogen) [reply code/modules/shuttle/special.dm]: Yeah. Slimes can't drag. Their drag proc is overridden. I wish there was an easier way to identify mobs that can't drag but aside from making a global list or changing how dragging is handled for all mobs I can't think of one.

Comment (actioninja) [reply code/modules/shuttle/special.dm]: Got it.

**Pull request merged**